plugins/default/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -0,0 +1,797 @@
+                        , let (testRange, resultRange) = testRanges test
+                              args = EvalParams (setups ++ [section]) _textDocument
+                              cmd' =
+                                (cmd :: Command)
+                                    { _arguments = Just (List [toJSON args])
+                                    , _title =
+                                        if trivial resultRange
+                                            then "Evaluate..."
+                                            else "Refresh..."
+                                    }
+                        ]
+
+                perf "tests" $
+                    dbg "Tests" $
+                        unwords
+                            [ show (length tests)
+                            , "tests in"
+                            , show (length nonSetups)
+                            , "sections"
+                            , show (length setups)
+                            , "setups"
+                            , show (length lenses)
+                            , "lenses."
+                            ]
+
+                return $ List lenses
+  where
+    trivial (Range p p') = p == p'
+
+evalCommandName :: CommandId
+evalCommandName = "evalCommand"
+
+evalCommand :: PluginCommand
+evalCommand = PluginCommand evalCommandName "evaluate" runEvalCmd
+
+-- |Specify the test section to execute
+data EvalParams = EvalParams
+    { sections :: [Section]
+    , module_ :: !TextDocumentIdentifier
+    }
+    deriving (Eq, Show, Generic, FromJSON, ToJSON)
+
+runEvalCmd :: CommandFunction EvalParams
+runEvalCmd lsp st EvalParams{..} =
+    let dbg = logWith st
+        perf = timed dbg
+        cmd = do
+            let tests = testsBySection sections
+
+            let TextDocumentIdentifier{_uri} = module_
+            fp <- handleMaybe "uri" $ uriToFilePath' _uri
+            let nfp = toNormalizedFilePath' fp
+            mdlText <- moduleText lsp _uri
+
+            session <- runGetSession st nfp
+
+            (ms, _) <-
+                liftIO $
+                    runAction "runEvalCmd.getModSummary" st $
+                        use_ GetModSummary nfp
+
+            now <- liftIO getCurrentTime
+
+            let modName = moduleName $ ms_mod ms
+                thisModuleTarget =
+                    Target
+                        (TargetFile fp Nothing)
+                        False
+                        (Just (textToStringBuffer mdlText, now))
+
+            -- Setup environment for evaluation
+            hscEnv' <-
+                withSystemTempFile (takeFileName fp) $ \logFilename logHandle ->
+                    ExceptT $
+                        (either Left id <$>)
+                            . gStrictTry
+                            $ evalGhcEnv (hscEnvWithImportPaths session) $
+                                do
+                                    env <- getSession
```

Comment:
Curious choice of indentation. Not only this PR is 60 files long, it's also 200 columns wide!

Reply:
beauty is in the eye of the beholder :-) 

I have squashed those 'header' lines 350-356 in a single line (that is now 182 columns wide!) so that we can fit the code under it into less than 100 columns.